AmplifyPlugins/Core/AWSPluginsCore/Model/Support/QueryPredicate+GraphQL.swift

```diff
@@ -28,6 +44,18 @@ extension QueryPredicate {
     }
 }
 
+extension String {
+
+    /// Deserialize the JSON string converted with `QueryPredicate.toGraphQLFilterString()` to `[String: Any]` JSON
+    public func toGraphQLFilterJson() throws -> [String: Any] {
```

Comment:
> This should not be a public extension.

isn't this public for DataStore to turn it back into GraphQLFilter? that is unless we change the GraphQLRequest extension methods to take in graphQLFilterJSON String and then internally do `graphQLFilterJSON.toGraphQLFiilter()`

Reply:
My point is that this is a generic String -> JSON function, added as an extension to String. Also the semantics are not entire correct. For example: `"{a:1}". toGraphQLFilterJson()` would return a valid JSON as `[String: Any]` but it's not a valid GraphQL filter.

Therefore, if the extension is `toJSON()`, that could be valid. Then the discussion  is if we should expose that extension to customers or not (it seems outside of the scope of Amplify to add such extension).

Reply:
taking a look at this to see if we can reduce the access level or at least move it under a static function of a different class rather than extension on String

Reply:
i agree, the string extension is incorrect in this case, and is prone to errors, and don't think we want to take on the `toJSON()` implementation. Updated to `GraphQLFilterConverter` with static methods for turning into a GraphQLFilter JSON and back to a GraphQLFilter